packages/peregrine/lib/talons/CartPage/ProductListing/useProductListing.js

```diff
@@ -0,0 +1,30 @@
+import { useQuery } from '@apollo/react-hooks';
+
+import { useCartContext } from '../../../context/cart';
+import { useEffect } from 'react';
+
+export const useProductListing = props => {
+    const { query } = props;
+
+    const [{ cartId }] = useCartContext();
+
+    const { data, error, loading } = useQuery(query, {
```

Comment:
You're going to want to change this to a lazy query since `cartId` can change. Will this work if you sign out/sign in?

Reply:
However, I'm not sure if that's necessary as cartId should update and the `useQuery` hook should update appropriately...

Reply:
Ok, so this will work correctly IF `cartId` is present, but you do need to make sure you aren't calling the query when `cartId` is undefined. After adding an item, go to the page and sign in/out. You should see a console error for graphql related to missing `cartId`.

Reply:
I was comfortable having error state === empty cart state, but agree its best to save the network request we know will fail. Added in f622664.